date adapter: add hack to parse ISO8601 on Android

ttaylorr opened the pull request:
Currently, Android's implementation of the `SimpleDAteFormat` class does not
include the format char `X`. On desktop clients, the wildcard `X` token was
being used to ignore the GMT zone signifer `Z`.

This works fine on the desktop, since GMT just means +00:00. On Android,
however, an exception is thrown.

As a temporary workaround, the un-parsable 'Z' is replaced with "+0000", which
is parsable with the `Z` delimiter on both Android and Windows.

GraphicalBear commented:
Just a reworked pull request with cleaner code.

ttaylorr commented on src/main/java/pro/beam/api/resource/chat/events/data/IncomingMessageData.java:
```diff
@@ -38,8 +40,10 @@ public String getMessage() {
         public String url;
         public String data;
         public String path;
+        public String text;
```
Hm... I wonder if we could get the text into `data` instead of adding another member var.

GraphicalBear replied:
Well, it's the way it currently is, merely because of the fact that the JSONObject received has the element named as "text" if the JSONElement were renamed to data then it should work fine I believe. that's a server side thing however.

GraphicalBear replied:
I mean the JSONObject could be reprocessed in the BeamChatConnectable class, but that seems inefficient when a simple name change to the json itself would solve the problem

ttaylorr commented:
Thanks so much for this!  It looks fine, aside from my one comment, then I'd be happy to merge.

GraphicalBear commented:
Sweet Taylor, thanks!

ttaylorr committed:
date adapter: add hack to parse ISO8601 on Android

Currently, Android's implementation of the `SimpleDAteFormat` class does not
include the format char `X`. On desktop clients, the wildcard `X` token was
being used to ignore the GMT zone signifer `Z`.

This works fine on the desktop, since GMT just means +00:00. On Android,
however, an exception is thrown.

As a temporary workaround, the un-parsable 'Z' is replaced with "+0000", which
is parsable with the `Z` delimiter on both Android and Windows.
